### PR TITLE
feat: grpc connection pool

### PR DESCRIPTION
**Pull request by Arqu**

### CODE REVIEW

File: iroh-rpc-types/src/macros.rs [hunk 1]
```diff
@@ -146,7 +147,8 @@ macro_rules! proxy {
                             #[cfg(feature = "grpc")]
                             Self::Grpc { client, .. } => {
                                 let req = iroh_metrics::req::trace_tonic_req(req);
-                                let mut c = client.clone();
+                                // TODO(pool): client.get().clone()
+                                let mut c = client.get();
```
Comment (Arqu) [inline iroh-rpc-types/src/macros.rs:151]: @dignifiedquire this line :(

File: iroh-rpc-types/src/connection_pool.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        // conn.execute_batch("").map_err(Into::into)
+        Ok(())
+    }
+
+    fn has_broken(&self, _: &mut Self::Connection) -> bool {
+        false
+    }
+}
+
+quick_error! {
```
Comment (dignifiedquire) [inline iroh-rpc-types/src/connection_pool.rs:71]: should use `thiserror`, `quick_error` is pretty outdated

File: iroh-metrics/Cargo.toml [hunk 3]
```diff
@@ -29,6 +29,7 @@ lazy_static = "1.4.0"
 async-trait = "0.1.57"
 console-subscriber = { version = "0.1.7", optional = true }
 paste = "1.0.9"
+tracing-appender = "0.2"
```
Comment (Arqu) [inline iroh-metrics/Cargo.toml:32]: need to drop this before merge.

**Comment by Arqu** [comment]
We got an impl here https://github.com/n0-computer/iroh/pull/404 which is higher up. Ideally we can fix this up and use this instead.

**Comment by b5** [comment]
closing, as we're going with the direction outlined in #459

**Pull request closed without merging**